### PR TITLE
Mandiant Advantage Threat Intelligence: Bugfix - Bearer Token and Indicator Search

### PR DESCRIPTION
**Pull request by chrishultin**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Fix issue where the Splunk SOAR connector was unable to retrieve a bearer token from Mandiant
- Fix issue when `indicator lookup` retrieves no results

---
Thanks for contributing!

### CODE REVIEW

**Review by tonyc-phantom — APPROVED** [review]

**Comment by dhwanis-crest** [comment]
@chrishultin, the release notes needs to be added in the unreleased.md file and no need to create md file for specific version. It will automatically be created once the MR gets merged. Also, for bug-fixes we update minor version as per the standards and not the patch version.

File: mandiantadvantagethreatintelligence_connector.py [hunk 1]
```diff
@@ -173,6 +175,10 @@ def _handle_test_connectivity(self, param, action_result):
         test_headers = {
             "Accept": "application/json"
         }
+        token_status = self._get_bearer_token(param, force=True)
+        if not token_status:
+            return action_result
```
Comment (achandya-crest) [inline mandiantadvantagethreatintelligence_connector.py]: Please update the return statement to something like: return action_result.set_status(phantom.APP_ERROR, "Test Connectivity Failed")

File: mandiantadvantagethreatintelligence_connector.py [hunk 2]
```diff
@@ -214,6 +220,14 @@ def _handle_indicator_lookup(self, param, action_result):
             self.save_progress("Error getting indicator info")
             return action_result.set_status(phantom.APP_ERROR, "Error getting indicator info")
 
+        if not response.get("indicators"):
```
Comment (achandya-crest) [inline mandiantadvantagethreatintelligence_connector.py]: Only one condition would suffice for no indicators present:

if not response.get("indicators", []):
    self.save_progress("No indicators retrieved from Mandiant")
    return action_result.set_status(phantom.APP_SUCCESS)

**Comment by chrishultin** [comment]
> @chrishultin, the release notes needs to be added in the unreleased.md file and no need to create md file for specific version. It will automatically be created once the MR gets merged. Also, for bug-fixes we update minor version as per the standards and not the patch version.

Should I update the "app_version" manually (to 1.1.0)?

**Comment by achandya-crest** [comment]
@chrishultin, I have added changes for the redundant conditions in lookup indicator and also updated the minor app version

**Review by ishans-crest — APPROVED** [review]

**Pull request merged**